os/board/rtl8730e/src/rtl8730e_st7789_lcd.c

```diff
@@ -0,0 +1,108 @@
+/****************************************************************************
+ *
+ * Copyright 2023 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ ****************************************************************************/
+/****************************************************************************
+ * Included Files
+ ****************************************************************************/
+
+#include <tinyara/config.h>
+/*
+#include <stdbool.h>
+#include <stdio.h>
+#include <debug.h>
+#include <assert.h>
+#include <errno.h>
+
+#include <tinyara/irq.h>
+#include <tinyara/audio/ndp120.h>
+#include <tinyara/gpio.h>
+#include "gpio_irq_api.h"
+#include <tinyara/audio/ndp120.h>
+*/
+#include "objects.h"
+#include "PinNames.h"
+#include "gpio_api.h"
+
+
+#define LCD_DEV_NO 0
+/* SPI frequency */
+
+#define LCD_SPI_PORT 1
+
+/****************************************************************************
+ * Name:  board_lcd_getdev
+ *
+ * Description:
+ *   Return a a reference to the LCD object for the specified LCD.  This
+ *   allows support for multiple LCD devices.
+ *
+ ****************************************************************************/
+
+struct lcd_dev_s *board_lcd_getdev(int devno)
+{
+
+	FAR struct spi_dev_s *spi = up_spiinitialize(LCD_SPI_PORT);
+
+  	struct lcd_dev_s *g_lcd = st7789_lcdinitialize(spi);
+  	if (!g_lcd)
+    	{
+		lldbg("ERROR: Failed to bind SPI port to LCD %d\n",devno);
+    	}
+  	else
+    	{
+      		lldbg("SPI1 port bound to LCD %d\n", devno);
+      		return g_lcd;
+    	}
+  	return NULL;
+}
+
+void board_lcd_initialize(void)
```

Comment:
1. please refer to how another driver initialized. you can refer to artik055_alc5658.c & audio.c
2. rename to rtl8730e_st7789.c
3. rename oo rtl8730_st7789_initialize this need to be called once in rtl8730e_boot.c
4. so above board_lcd_getdev should be removed.

option 2, 3 is acceptable if it should handle some specific operation like a gpio interrupt.
otherwise you can call as a below in rtl8730e_boot.c
```
#if defined(CONFIG_LCD) && defined(CONFIG_LCD_ST7789) //CONFIG_LCD is necessary??
	struct spi_dev_s *spi;
	struct lcd_dev_s *dev;
	spi = up_spiinitialize(x);
	dev = st7789_lcdinitialize(spi);
	lcddev_register(dev);
#endif
```
code line 75 to 103 seems that option 2, 3 is necessary, IMO.